lib/client/sse_client.dart

```diff
@@ -95,24 +95,16 @@ class SseClient extends StreamChannelMixin<String> {
     close();
   }
 
-  final _messages = StreamController<dynamic>();
-
-  void _onOutgoingMessage(dynamic message) async {
-    _messages.add(message);
-  }
-
-  void _startPostingMessages() async {
-    await for (var message in _messages.stream) {
-      try {
-        await HttpRequest.request(_serverUrl,
-            method: 'POST',
-            sendData: jsonEncode(message),
-            withCredentials: true);
-      } on JsonUnsupportedObjectError catch (e) {
-        _logger.warning('Unable to encode outgoing message: $e');
-      } on ArgumentError catch (e) {
-        _logger.warning('Invalid argument: $e');
-      }
+  void _onOutgoingMessage(String message) async {
+    String encodedMessage;
+    try {
+      encodedMessage = jsonEncode(message);
+    } on JsonUnsupportedObjectError catch (e) {
+      _logger.warning('Unable to encode outgoing message: $e');
+    } on ArgumentError catch (e) {
+      _logger.warning('Invalid argument: $e');
     }
+    await HttpRequest.request('$_serverUrl&messageId=${++_lastMessageId}',
```

Comment:
We should keep track of when a request failed and retry sending a a dummy "ack" message to the server indicating a failure. that way if there is a failure only the particular failed message is skipped rather than the server hanging indefinitely.

Reply:
I don't think we need to send a dummy "ack" but we should decrement the message ID counter.